go/pkg/bertymessenger/handlers.go

```diff
@@ -435,76 +520,154 @@ func handleAppMessage(svc *service, gpk string, gme *bertytypes.GroupMessageEven
+			if err := svc.db.Create(i).Error; err != nil {
+				return err
+			}
+
+			if err := svc.dispatcher.StreamEvent(StreamEvent_TypeInteractionUpdated, &StreamEvent_InteractionUpdated{i}); err != nil {
+				return err
+			}
 
-			// send ack
+			if !isMe {
+				svc.logger.Debug("sending ack", zap.String("target", cid))
+
+				// send ack
+
+				// Don't send ack if message is already acked to prevent spam in multimember groups
+				// Maybe wait a few seconds before checking since we're likely to receive the message before any ack
+				/*var ei Interaction
+				err := svc.db.Where("cid = ? AND acknowledged = ?", cid, true).Take(&ei).Error
+				if err == gorm.ErrRecordNotFound {
+					return nil
+				} else if err != nil {
+					return err
+				}*/
 
-			// Don't send ack if message is already acked to prevent spam in multimember groups
-			// Maybe wait a few seconds before checking since we're likely to receive the message before any ack
-			/*var ei Interaction
-			err := svc.db.Where("cid = ? AND acknowledged = ?", cid, true).Take(&ei).Error
-			if err == gorm.ErrRecordNotFound {
+				amp, err := AppMessage_TypeAcknowledge.MarshalPayload(&AppMessage_Acknowledge{Target: cid})
+				if err != nil {
+					return err
+				}
+				req := bertytypes.AppMessageSend_Request{
+					GroupPK: gme.GetEventContext().GetGroupPK(),
+					Payload: amp,
+				}
+				_, err = svc.protocolClient.AppMessageSend(svc.ctx, &req)
+				if err != nil {
+					return err
+				}
+			}
+		case AppMessage_TypeSetUserName:
+			if isMe {
 				return nil
-			} else if err != nil {
-				return err
-			}*/
+			}
 
-			amp, err := AppMessage_TypeAcknowledge.MarshalPayload(&AppMessage_Acknowledge{Target: cid})
-			if err != nil {
-				return err
+			if device == nil {
+				return errcode.ErrMissingInput
 			}
-			req := bertytypes.AppMessageSend_Request{
-				GroupPK: gme.GetEventContext().GetGroupPK(),
-				Payload: amp,
+
+			payload := amPayload.(*AppMessage_SetUserName)
+
+			member := Member{
+				PublicKey:             device.GetOwnerPublicKey(),
+				ConversationPublicKey: gpk,
+				DisplayName:           payload.GetName(),
 			}
-			_, err = svc.protocolClient.AppMessageSend(svc.ctx, &req)
+			err := svc.db.
+				Clauses(clause.OnConflict{
+					Columns:   []clause.Column{{Name: "public_key"}},
+					DoUpdates: clause.AssignmentColumns([]string{"display_name"}),
+				}).
+				Create(&member).
+				Error
 			if err != nil {
 				return err
 			}
+
+			return svc.dispatcher.StreamEvent(StreamEvent_TypeMemberUpdated, &StreamEvent_MemberUpdated{Member: &member})
+
+		default:
+			svc.logger.Warn("unsupported app message type", zap.String("type", amt.String()))
 		}
 
-	default:
-		svc.logger.Warn("unsupported app message type", zap.String("type", amt.String()))
+		return nil
+	}
+
+	if isMe || conv.GetType() == Conversation_ContactType {
+		return next(nil)
+	}
+
+	existingDevice := &Device{}
+	err = svc.db.Where(&Device{PublicKey: dpk}).Take(existingDevice).Error
```

Comment:
```suggestion
	existingDevice := &Device{PublicKey: dpk}
	err = svc.db.Where(existingDevice).Take(existingDevice).Error
```

I prefer writing this way, don't know about you

Reply:
I prefer my way because it make explicit that the object passed to `.Where(` is only a source